homeassistant/components/thinkingcleaner/switch.py

```diff
@@ -9,19 +11,28 @@
 from homeassistant.components.switch import PLATFORM_SCHEMA
 from homeassistant.const import CONF_HOST, STATE_OFF, STATE_ON
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.helpers.entity import ToggleEntity, ToggleEntityDescription
 
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(milliseconds=100)
 
 MIN_TIME_TO_WAIT = timedelta(seconds=5)
 MIN_TIME_TO_LOCK_UPDATE = 5
 
-SWITCH_TYPES = {
-    "clean": ["Clean", None, None],
-    "dock": ["Dock", None, None],
-    "find": ["Find", None, None],
-}
+SWITCH_TYPES: tuple[ToggleEntityDescription, ...] = (
```

Comment:
These should be using the `SwitchEntityDescription`s (and the entity should be extending `SwitchEntity` and not `ToggleEntity`.

So, at this point, it makes sense to use `ToggleEntityDescription`.